scanpy/plotting/preprocessing.py

```diff
@@ -41,6 +41,12 @@ def filter_genes_dispersion(result, log=False, save=None, show=None):
             pl.ylim(y_min, 1.05*np.max(dispersions))
         pl.legend()
         pl.xlabel(('$log_{10}$ ' if False else '') + 'mean expression of gene')
-        pl.ylabel(('$log_{10}$ ' if False else '') + 'dispersion of gene'
-                  + (' (normalized)' if id == 0 else ' (not normalized)'))
-    utils.savefig_or_show('filter_genes_dispersion', show=show, save=save)
+
+        if id == 0:
+            pl.ylabel(('$log_{10}$ ' if False else '') + 'dispersion of gene'
+                  + (' (normalized)'))
+            utils.savefig_or_show('filter_genes_dispersion_normalized', show=show, save=save)
+        else:
+            pl.ylabel(('$log_{10}$ ' if False else '') + 'dispersion of gene'
+                  + (' (not normalized)'))
+            utils.savefig_or_show('filter_genes_dispersion', show=show, save=save)
```

Comment:
Could you please eliminate repetition here? The only difference is `'filter_genes_dispersion_normalized'` vs `'filter_genes_dispersion'`.

You can simply do `label = 'filter_genes_dispersion{}'.format('_normalized' if id == 0 else '')` and use the variable in the block without repeating the whole thing. Also makes for a smaller diff.